### ISSUE TITLE
Idea for ship upgrades.

### ISSUE THREAD

**Issue by icydee** [plainblack/Lacuna-Server-Open#369]
Consider a new class of plan (let's call them blueprints).

Blueprints would allow single ships to be upgraded, for example speed, hold-size, stealth, combat, berth-level.

Options to obtain blueprints could include some of :- 

  * Like plans, via excavators
  * Only by theft from an AI
  * Only by sale from an AI (Delambert)
  * 'special events' (not sure what this means)

The implementation could be kept fairly simple. We just need a UI to link a blueprint with a ship and the upgrade would take effect immediately and the blueprint would be deleted. This implementation would imply that multiple blueprints could be applied to the same ship thus creating very high spec ships (I don't see why this would be a problem if blueprints were fairly difficult to obtain).

If the Del had a stock of blueprints we could even let the Del sell higher spec ships in trade.

Question, where do blueprints come from? Would we want a new building that is (again difficult to obtain) related to blueprint production?

**Comment by Lacy101** [comment]
A new building of 4 plots like the SS lab upgradeble to lv30, With better blueprints getting avaidable per each level of the "Blueprint Manufactury"

**Comment by Imzogelmo** [comment]
Having them come from the Del seems like a good idea. Another idea would be to have some missions that give them (care taken so that the blueprints given from the uni levels are pertinent to the ships you're likely to be using at that level). 
I personally don't think a new building is needed, but if it were done that way, a 4x4 building like the SS lab would be a way to provide a tradeoff (plots vs. better ships) to make it not totally "obvious". However, no matter how big the building is, it would still be "obvious" to have at least one somewhere. I don't know how it could really be mitigated to not be an arms race if it were a building. Having it through a less controllable pipeline like the AI or missions would prevent that, or at least slow it down, and it would encourage trade too.

**Comment by dmcbride** [comment]
> care taken so that the blueprints given from the uni levels are pertinent to the ships you're likely to be using at that level

Not sure - if you can get blueprints at lower levels for stuff higher level people can build, it might create another source of trade, like the LCOT GHI plans more or less used to be.  They wouldn't be solely given out this way, but maybe a bit asymmetrical?

**Comment by Imzogelmo** [comment]
Yeah, I mainly was worried about the other way around. I.E. blueprints for barges that need uni 30. Low to high is actually a good idea.
